diff --git a/BUILD.gn b/BUILD.gn
index 4d88fd67d86..5898483f0e9 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -99,6 +99,10 @@ group("create_sdk") {
   public_deps = [ "sdk:create_sdk" ]
 }
 
+group("libdart") {
+  deps = [ "runtime/bin:libdart" ]
+}
+
 group("create_platform_sdk") {
   public_deps = [ "sdk:create_platform_sdk" ]
 }
diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index f6c2c6625dc..f3986381da4 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -250,8 +250,8 @@ config("compiler") {
   }
 
   if (is_android || is_linux || is_mac || is_fuchsia) {
-    cflags += [ "-fPIE" ]
-    ldflags += [ "-fPIE" ]
+    cflags += [ "-fPIC" ]
+    ldflags += [ "-fPIC" ]
   }
 
   # Linux-specific compiler flags setup.
diff --git a/runtime/bin/BUILD.gn b/runtime/bin/BUILD.gn
index e757b71d483..aa10a453907 100644
--- a/runtime/bin/BUILD.gn
+++ b/runtime/bin/BUILD.gn
@@ -1110,3 +1110,49 @@ if (defined(is_linux) && is_linux && defined(is_asan) && is_asan &&
     }
   }
 }
+
+static_library("libdart") {
+  deps = [
+    ":standalone_dart_io",
+    "..:libdart_jit",
+    "../platform:libdart_platform_jit",
+    ":dart_snapshot_cc",
+    ":dart_kernel_platform_cc",
+    "//third_party/boringssl",
+    "//third_party/zlib",
+  ]
+  if (dart_runtime_mode != "release") {
+    deps += [ "../observatory:standalone_observatory_archive" ]
+  }
+
+  complete_static_lib = true
+
+  if (dart_use_tcmalloc) {
+    deps += [ "//third_party/tcmalloc" ]
+  }
+
+  include_dirs = [
+    "..",
+    "//third_party",
+  ]
+
+  sources = [
+    "builtin.cc",
+    "error_exit.cc",
+    "error_exit.h",
+    "vmservice_impl.cc",
+    "vmservice_impl.h",
+    "snapshot_utils.cc",
+    "snapshot_utils.h",
+    "gzip.cc",
+    "gzip.h",
+    "dfe.cc",
+    "dfe.h",
+    "loader.cc",
+    "loader.h",
+    "dart_embedder_api_impl.cc",
+  ]
+  if (dart_runtime_mode == "release") {
+    sources += [ "observatory_assets_empty.cc" ]
+  }
+}
