diff --git a/BUILD.gn b/BUILD.gn
index 233a0900e55..5010af8fc55 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -99,6 +99,10 @@ group("create_sdk") {
   public_deps = [ "sdk:create_sdk" ]
 }
 
+group("libdart") {
+  deps = [ "runtime/bin:libdart" ]
+}
+
 group("create_platform_sdk") {
   public_deps = [ "sdk:create_platform_sdk" ]
 }
diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index fb45d1ec551..9ccb1a27f4a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -269,8 +269,9 @@ config("compiler") {
       cflags += [ "-fPIC" ]
       ldflags += [ "-fPIC" ]
     } else {
-      cflags += [ "-fPIE" ]
-      ldflags += [ "-fPIE" ]
+      # dart_shared_library overriden - we need -fPIC to properly build
+      cflags += [ "-fPIC" ]
+      ldflags += [ "-fPIC" ]
     }
   }
 
diff --git a/runtime/bin/BUILD.gn b/runtime/bin/BUILD.gn
index 91ebf8feb65..4ecf43807d4 100644
--- a/runtime/bin/BUILD.gn
+++ b/runtime/bin/BUILD.gn
@@ -1143,3 +1143,47 @@ if (defined(is_linux) && is_linux && defined(is_asan) && is_asan &&
     }
   }
 }
+
+static_library("libdart") {
+  deps = [
+    ":standalone_dart_io",
+    "..:libdart_jit",
+    "../platform:libdart_platform_jit",
+    ":dart_snapshot_cc",
+    ":dart_kernel_platform_cc",
+    "//third_party/boringssl",
+    "//third_party/zlib",
+  ]
+  if (dart_runtime_mode != "release") {
+    deps += [ "../observatory:standalone_observatory_archive" ]
+  }
+
+  complete_static_lib = true
+
+  include_dirs = [
+    "..",
+    "//third_party",
+  ]
+
+  sources = [
+    "builtin.cc",
+    "error_exit.cc",
+    "error_exit.h",
+    "vmservice_impl.cc",
+    "vmservice_impl.h",
+    "snapshot_utils.cc",
+    "snapshot_utils.h",
+    "gzip.cc",
+    "gzip.h",
+    "dartdev_isolate.cc",
+    "dartdev_isolate.h",
+    "dfe.cc",
+    "dfe.h",
+    "loader.cc",
+    "loader.h",
+    "dart_embedder_api_impl.cc",
+  ]
+  if (dart_runtime_mode == "release") {
+    sources += [ "observatory_assets_empty.cc" ]
+  }
+}
